mage_ai/streaming/sources/mongodb.py

```diff
@@ -0,0 +1,69 @@
+from dataclasses import dataclass
+from typing import Any, Callable, Mapping, Optional
+
+from bson.timestamp import Timestamp
+from pymongo import MongoClient
+from pymongo.typings import _Pipeline
+
+from mage_ai.shared.config import BaseConfig
+from mage_ai.streaming.sources.base import BaseSource
+
+
+@dataclass
+class MongoDBConfig(BaseConfig):
+    connection_str: str
+    database: str
+    collection: str = None
+    batch_size: Optional[int] = 100
+    pipeline: Optional[_Pipeline] = None
+    operation_time: Optional[Timestamp] = None
+    start_after: Optional[Mapping[str, Any]] = None
+
+
+class MongoSource(BaseSource):
+    config_class = MongoDBConfig
+
+    def init_client(self):
+        connection_str = self.config.connection_str
+        database = self.config.database
+        collection = self.config.collection
+        batch_size = self.config.batch_size
+        pipeline = self.config.pipeline
+        operation_time = self.config.operation_time
+        start_after = self.config.start_after
+        self._print(f"Starting to MongoDB Stream for {database}")
+        self._print(f"You can use this collection {collection}")
+        self._print(f"You can use {connection_str}")
+        self._print(f"You can use {batch_size}")
+        self._print(f"You can use {pipeline}")
+        self._print(f"You can use {operation_time}")
+        self._print(f"You can use {start_after}")
+
+    def read(self, handler: Callable):
+        pass
+
+    def batch_read(self, handler: Callable):
+        self._print("Start getting message for MongoDB streaming.")
+        try:
+            client = MongoClient(self.config.connection_str)
+            db = client.get_database(self.config.database)
+            parsed_messages = []
+            if self.config_class.collection:
+                watch_args = {'batch_size': self.config.batch_size}
+                if self.config_class.pipeline:
+                    watch_args['pipeline'] = self.config.pipeline
+                if self.config_class.operation_time:
+                    watch_args['operation_time'] = self.config.operation_time
+                if self.config_class.start_after:
+                    watch_args['start_after'] = self.config.start_after
+                with db[self.config_class.collection].watch(**watch_args) as stream:
```

Comment:
The batch_read method should keep fetching changes from the mongodb stream and use handler to process them.

Reply:
Yes I understand that. watch_args are parameters of MongoCDC such as where I started to updating data, pipeline is query for getting streaming. You can look at

pipeline = [
    {"$match": {"fullDocument.username": "alice"}},
    {"$addFields": {"newField": "this is an added field!"}},
]
cursor = db.inventory.watch(pipeline=pipeline)
next(cursor)



in https://www.mongodb.com/docs/manual/changeStreams/

If you say it doesn't needed it I delete this parameter.

Reply:
No, I mean it should fetch a batch from the stream and use handler to process the batch. Then fetch next batch and process next batch.
You currently fetch all the changes first and call handler at then end. I don't know whether that works.

Reply:
I understand that. You say that, write this parameter once and this parameter can use for this. Am I wrong?

Reply:
I didn't say anything about the parameter.